### PR TITLE
got watchlist data fetch working

### PR DESCRIPTION
**Pull request by lucaimbrogno**
Let me know if I did this stuff right. 

I created the Lists view and a separate Watchlist component in case we want to have multiple watchlists within the Lists view in the future.

### CODE REVIEW

File: packages/dashboard/src/components/lists/Watchlist.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  Dropdown,
+  DropdownToggle,
+  DropdownMenu,
+  DropdownItem,
+  Card,
+  CardHeader,
+  CardBody,
+} from 'shards-react'
+
+
```
Comment (alexanderMontague) [inline packages/dashboard/src/components/lists/Watchlist.js]: dont worry about formatting now, I'll run that script after. But deffs look into vscode auto formatting so the formatting diffs dont dirty up PRS

File: packages/dashboard/src/components/lists/Watchlist.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+
+const Watchlist = ({
```
Comment (alexanderMontague) [inline packages/dashboard/src/components/lists/Watchlist.js]: Are we swapping these around? ie this be the `List` component and the main view be `Watchlist`

File: packages/dashboard/src/components/lists/Watchlist.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+  user,
+  historicQuotes,
+  globalSelectedAccount,
+}) => {
+  
+
+  
+
+  return (
+    <>
```
Comment (alexanderMontague) [inline packages/dashboard/src/components/lists/Watchlist.js]: ik theres nothing here yet but if you didnt already know, you don't need fragments if you already have a surrounding component like `<Container .. />`. They don't get rendered in the DOM but useful for grouping top level components as you can only return 1 element.

File: packages/dashboard/src/redux/reducers/trade.js [hunk 4]
```diff
@@ -70,6 +74,24 @@ const tradeState = (prevState = initialState, { type, payload }) => {
...
+        isWatchlistLoading: true,
+      }
+    }
+
+    case GET_WATCHLIST_RESPONSE: {
+      return {
+        ...prevState,
+        isWatchlistLoading: false,
+        watchlist: {
+          ...prevState.watchlist,
```
Comment (alexanderMontague) [inline packages/dashboard/src/redux/reducers/trade.js]: I don't think we need this. I only did this on  `historicQuotes` so we keep all previous historic quotes and only overwrite the new one we get back.

File: packages/dashboard/src/redux/reducers/trade.js [hunk 5]
```diff
@@ -70,6 +74,24 @@ const tradeState = (prevState = initialState, { type, payload }) => {
...
+          ...payload.data,
```
Comment (alexanderMontague) [inline packages/dashboard/src/redux/reducers/trade.js]: Just making sure you want this to happen.
In your saga you dispatch the response action with the request payload which looks like:
```
{
    "securities": [
        { ... },
        { ... },
        ...
      ],
      total_count: 5
}
```


That means the watchlist object will get set to this data, but in that case you can just overwrite the whole thing.
```suggestion
         return {
          ...prevState,
          isWatchlistLoading: false,
          watchlist: payload.data
        }
```
This suggestion is for the whole returned object

File: packages/dashboard/src/redux/sagas/trade.js [hunk 6]
```diff
@@ -19,6 +19,22 @@ function* getAccountHistory({ payload }) {
...
 
+function* getAccountWatchlist({ payload }) {
+  let watchlistData
+  try {
+    watchlistData = yield getWatchlist(payload)
+  } catch (error) {
+    watchlistData = createResponse(
+      error.response?.status,
+      error.message,
+      null,
```
Comment (alexanderMontague) [inline packages/dashboard/src/redux/sagas/trade.js]: Probably going to want this to be `{}` as if this errors you ultimately set the `watchlist` state to this. When you start doing checks on the watchlist data, if its sometimes an object and sometimes null that causes headaches.

File: packages/dashboard/src/views/Lists.js [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+const Lists = ({
+  getWatchlist,
+  watchlist,
+  user,
+  isLoggedIn
+}) => {
+
+  useEffect(() => {
+    if (isLoggedIn) {
+      getWatchlist({tokens: JSON.stringify(user.tokens)})
```
Comment (alexanderMontague) [inline packages/dashboard/src/views/Lists.js]: ✅

**Review by alexanderMontague — COMMENTED** [review]
Few things but gj this guys lookin like a real react dev

File: packages/dashboard/src/routes/index.js [hunk 8]
```diff
@@ -13,6 +13,7 @@ import Errors from '../views/Errors'
 import ComponentsOverview from '../views/ComponentsOverview'
 import Tables from '../views/Tables'
 import BlogPosts from '../views/BlogPosts'
+import Lists from '../views/Lists'
```
Comment (alexanderMontague) [inline packages/dashboard/src/routes/index.js]: Should this be Watchlist? From `../views/Watchlists`

File: packages/dashboard/src/views/Watchlists.js [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+            <PageTitle
+                sm="4"
+                title="Watchlists"
+                className="text-sm-left"
+            />
+        </Row>
+        <button onClick={checkWatchlist}>
+          check wl
+        </button>
+        <Watchlist />
```
Comment (alexanderMontague) [inline packages/dashboard/src/views/Watchlists.js]: `<List />` ?

**Review by alexanderMontague — COMMENTED** [review]
Might need to make these changes

File: packages/dashboard/src/views/Watchlists.js (hunk 9 above)
Comment (lucaimbrogno) [reply packages/dashboard/src/views/Watchlists.js]: woops forgot

**Review by alexanderMontague — APPROVED** [review]

**Pull request merged**